File: 38-count-and-say/count-and-say.cpp
class Solution {
public:
    string countAndSay(int n) {
        string ans = "1";
        for (int j = 1; j < n; j++) {
            string tem = "";
            int cnt = 1;
            for (int i = 1; i < ans.size(); i++) {
                if (ans[i] != ans[i - 1]) {
                    tem += to_string(cnt);
                    tem += ans[i - 1];
                    cnt = 1;
                } else {
                    cnt++;
                }
            }
            tem += to_string(cnt);
            tem += ans.back();
            ans = tem;
        }
        return ans;
    }
};

File: 38-count-and-say/count-and-say.py
class Solution:
    def countAndSay(self, n: int) -> str:
        ans = "1"
        for j in range(1 , n) :
            tem = ""
            cnt = 1
            for i in range(1 ,len(ans)) :
                if ans[i] != ans[i-1] :
                    tem += str(cnt)
                    tem += str(ans[i-1])
                    cnt = 1
                else :
                    cnt += 1
            tem += str(cnt)
            tem += str(ans[-1])
            ans =  tem
        return ans

